config.go

```diff
@@ -42,37 +42,59 @@ func (v Volume) String() string {
 	return ""
 }
 
-type Pairs []Pair
-type Pair struct {
-	Name  *string `yaml:"name,omitempty"`
+type Envs []Env
+type Env struct {
+	Name  *string `yaml:"name,omitempty" jsonschema:"pattern=^[-._a-zA-Z][-._a-zA-Z0-9]*$"`
 	Value *string `yaml:"value"`
 }
 
-func (e Pair) String() string {
+func (e Env) String() string {
 	if e.Name == nil && e.Value != nil {
 		match := regWholeSecret.FindStringSubmatch(*e.Value)
 		if len(match) == 2 {
-			return fmt.Sprintf("All key=value pairs from Secret \"%s\"", match[1])
+			return fmt.Sprintf("Env key=value pairs from Secret \"%s\"", match[1])
 		}
 		match = regWholeConfigMap.FindStringSubmatch(*e.Value)
 		if len(match) == 2 {
-			return fmt.Sprintf("All key=value pairs from ConfigMap \"%s\"", match[1])
+			return fmt.Sprintf("Env key=value pairs from ConfigMap \"%s\"", match[1])
 		}
 	} else if e.Name != nil && e.Value != nil {
 		match := regKeyFromSecret.FindStringSubmatch(*e.Value)
 		if len(match) == 3 {
-			return fmt.Sprintf("Entry \"%s\" with value set from key \"%s\" from Secret \"%s\"", *e.Name, match[2], match[1])
+			return fmt.Sprintf("Env \"%s\" with value set from key \"%s\" from Secret \"%s\"", *e.Name, match[2], match[1])
 		}
 		match = regKeyFromConfigMap.FindStringSubmatch(*e.Value)
 		if len(match) == 3 {
-			return fmt.Sprintf("Entry \"%s\" with value set from key \"%s\" from ConfigMap \"%s\"", *e.Name, match[2], match[1])
+			return fmt.Sprintf("Env \"%s\" with value set from key \"%s\" from ConfigMap \"%s\"", *e.Name, match[2], match[1])
 		}
 		match = regLocalEnv.FindStringSubmatch(*e.Value)
 		if len(match) == 2 {
-			return fmt.Sprintf("Entry \"%s\" with value set from local env variable \"%s\"", *e.Name, match[1])
+			return fmt.Sprintf("Env \"%s\" with value set from local env variable \"%s\"", *e.Name, match[1])
 		}
 
-		return fmt.Sprintf("Entry \"%s\" with value \"%s\"", *e.Name, *e.Value)
+		return fmt.Sprintf("Env \"%s\" with value \"%s\"", *e.Name, *e.Value)
+	}
+	return ""
+}
+
+type Labels []Label
+type Label struct {
+	// Key consist of optional prefix part (ended by '/') and name part
+	// Prefix part validation pattern: [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*
+	// Name part validation pattern: ([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]
+	Key   *string `yaml:"key" jsonschema:"pattern=^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\\/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$"`
```

Comment:
What I most love about RegEx in general is how easy they are to just take a quick glance at, and know exactly what they are matching. :sob: :rofl:

Reply:
lol 🔥

Reply:
I copypasted those two regexes directly from Kubernetes validation code. I tested the combined regex, so it should be ok 🤞